File: Store/test/Sales.Domain.Tests/Generators/OrderGenerator.cs
using System.Collections.Generic;
using System.Linq;
using System.Runtime.InteropServices.ComTypes;
using FsCheck;
using Sales.Domain.Model.Orders;

namespace Sales.Domain.Tests.Generators
{
    public class OrderGenerator
    {
        public static Arbitrary<Order> Generate()
        {
            return Arb.From(from id in Arb.Generate<PositiveInt>()
                    from items in OrderItemGenerator.Generate().Generator.ListOf()
                select new Order(id.Get, items.ToList()));
        }
    }
}

File: src/test/java/store/sales/domain/generators/OrderGenerator.java
package store.sales.domain.generators;

import com.pholser.junit.quickcheck.generator.GenerationStatus;
import com.pholser.junit.quickcheck.generator.Generator;
import com.pholser.junit.quickcheck.random.SourceOfRandomness;
import store.sales.domain.model.orders.Order;
import store.sales.domain.model.orders.OrderItem;

import java.util.ArrayList;
import java.util.List;

public class OrderGenerator extends Generator<Order> {

    public OrderGenerator() {
        super(Order.class);
    }

    @Override
    public Order generate(SourceOfRandomness random, GenerationStatus status) {
        return new Order(1, generateItems(random, status));
    }

    private List<OrderItem> generateItems(SourceOfRandomness random, GenerationStatus status) {
        OrderItemGenerator productGenerator = gen().make(OrderItemGenerator.class);
        List<OrderItem> products = new ArrayList<>();
        products.add(productGenerator.generate(random, status));
        return products;
    }
}
